src/ui/common/src/components/integrations/dialogs/kubernetesDialog.tsx

```diff
@@ -1,43 +1,102 @@
+import { Checkbox, FormControlLabel } from '@mui/material';
 import Box from '@mui/material/Box';
 import React from 'react';
+import { useEffect, useState } from 'react';
 
 import { KubernetesConfig } from '../../../utils/integrations';
+import { useAqueductConsts } from '../../hooks/useAqueductConsts';
 import { IntegrationTextInputField } from './IntegrationTextInputField';
 
 const Placeholders: KubernetesConfig = {
   kubeconfig_path: 'home/ubuntu/.kube/config',
   cluster_name: 'aqueduct',
+  use_same_cluster: 'false',
 };
 
 type Props = {
   onUpdateField: (field: keyof KubernetesConfig, value: string) => void;
   value?: KubernetesConfig;
+  apiKey: string;
 };
 
-export const KubernetesDialog: React.FC<Props> = ({ onUpdateField, value }) => {
+const { apiAddress } = useAqueductConsts();
+
+export const KubernetesDialog: React.FC<Props> = ({
+  onUpdateField,
+  value,
+  apiKey,
+}) => {
+  const [inK8sCluster, setInK8sCluster] = useState(false);
+  useEffect(() => {
+    if (!value?.use_same_cluster) {
+      onUpdateField('use_same_cluster', 'false');
+    }
+
+    const fetchEnvironment = async () => {
+      const environmentResponse = await fetch(`${apiAddress}/api/environment`, {
+        method: 'GET',
+        headers: {
+          'api-key': apiKey,
+        },
+      });
+
+      const responseBody = await environmentResponse.json();
+      setInK8sCluster(responseBody['inK8sCluster']);
+    };
+
+    fetchEnvironment().catch(console.error);
```

Comment:
neat, didn't know you could do this actually. I usually just wrap the async stuff with a try/catch statement and handle errors from inside the catch block.